devicetypes/smartthings/zwave-multi-metering-switch.src/zwave-multi-metering-switch.groovy

```diff
@@ -213,71 +246,85 @@ def ping() {
 	refresh()
 }
 
-def reset(endpoint = 1) {
-	log.debug "Resetting endpoint: ${endpoint}"
-	delayBetween([
-			secureEncap(zwave.meterV3.meterReset(), endpoint),
-			secureEncap(zwave.meterV3.meterGet(scale: 0), endpoint),
-			"delay 500"
-	], 500)
-}
-
 def childOnOff(deviceNetworkId, value) {
 	def switchId = getSwitchId(deviceNetworkId)
 	if (switchId != null) sendHubCommand onOffCmd(value, switchId)
 }
 
-def childRefresh(deviceNetworkId) {
-	def switchId = getSwitchId(deviceNetworkId)
-	if (switchId != null) sendHubCommand refresh(switchId)
+private onOffCmd(value, endpoint = 1) {
+	delayBetween([
+			secureEncap(zwave.basicV1.basicSet(value: value), endpoint),
+			secureEncap(zwave.basicV1.basicGet(), endpoint),
+			"delay 3000",
+			secureEncap(zwave.meterV3.meterGet(scale: 0), endpoint),
+			secureEncap(zwave.meterV3.meterGet(scale: 2), endpoint)
+	])
 }
 
-def childReset(deviceNetworkId) {
+private refreshBasicGetAll() {
+	childDevices.each { childRefreshBasicGet(it.deviceNetworkId) }
+	sendHubCommand refreshBasicGet()
+}
+
+def childRefreshBasicGet(deviceNetworkId) {
 	def switchId = getSwitchId(deviceNetworkId)
 	if (switchId != null) {
-		log.debug "Child reset switchId: ${switchId}"
-		sendHubCommand reset(switchId)
+		sendHubCommand refreshBasicGet(switchId)
 	}
 }
 
-def getSwitchId(deviceNetworkId) {
-	def split = deviceNetworkId?.split(":")
-	return (split.length > 1) ? split[1] as Integer : null
+def refreshBasicGet(endpoint = 1) {
```

Comment:
Instead of creating this separate method, can you just add a flag with a default argument to the definition of `refresh`?

Reply:
sure, fixed